packages/deployer/index.js

```diff
@@ -1,7 +1,40 @@
 const fs = require('fs');
 const path = require('path');
+const { extendConfig, extendEnvironment } = require('hardhat/config');
 
-['tasks', 'subtasks', 'extensions'].forEach((folder) =>
+extendConfig((config, userConfig) => {
```

Comment:
We can have extendConfig and extendEnvironment as independent files in extensions to have this index super slim and without any specific logic

Reply:
Yeah, I removed the extensions because I prefered to have all the global env defined on a single place (like [this](https://github.com/nomiclabs/hardhat-ts-plugin-boilerplate/blob/master/src/index.ts)), but as its probably going to get larger I will put it back as before.